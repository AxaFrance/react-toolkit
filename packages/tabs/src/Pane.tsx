import React from 'react';
import { withClassDefault, compose } from '@axa-fr/react-toolkit-core';

const DEFAULT_CLASSNAME = 'af-tabs__pane';

export type DivProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

interface PaneComponentProps extends DivProps {
  active: boolean;
  className?: string;
  children?: React.ReactNode;
}
const Pane: React.SFC<PaneComponentProps> = ({ children, className, active }) => {
  if(!active) {
    return null;
  }
  return (<div className={className}>{children}</div>);
};

const enchance = compose<PaneComponentProps, PaneComponentProps>(
  withClassDefault(DEFAULT_CLASSNAME)
)(Pane);

export default enchance;
