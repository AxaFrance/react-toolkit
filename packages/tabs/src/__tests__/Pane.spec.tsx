import React from 'react';
import { render } from '@testing-library/react';
import Pane from '../Pane';

describe('<Pane>', () => {
  it('should render component', () => {
    const { asFragment } = render(
      <Pane active>
        <div>Content</div>
      </Pane>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
