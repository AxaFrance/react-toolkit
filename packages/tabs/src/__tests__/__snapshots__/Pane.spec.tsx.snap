// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pane> should render component 1`] = `
<DocumentFragment>
  <div
    class="af-tabs__pane"
  >
    <div>
      Content
    </div>
  </div>
</DocumentFragment>
`;
